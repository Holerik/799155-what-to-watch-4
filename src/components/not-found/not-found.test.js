// not-found.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NotFound from './not-found.jsx';
import history from '../../history.js';
import {Router} from 'react-router-dom';

const mockStore = configureStore([]);

describe(`NotFound test`, () => {
  it(`NotFound component should render correctly`, () => {
    const store = mockStore({
      ERROR: {
        message: ``,
        show: false,
      },
    });

    const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <NotFound/>
          </Router>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
