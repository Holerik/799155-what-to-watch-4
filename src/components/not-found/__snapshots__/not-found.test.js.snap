// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound test NotFound component should render correctly 1`] = `
Array [
  <h1
    className="not-found"
  >
    404.
    <br />
    <br />
    <small>
      Page not found
    </small>
  </h1>,
  <div
    className="not-found"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Go to main page
    </a>
  </div>,
]
`;
